core/primitives/src/errors.rs

```diff
@@ -384,8 +384,8 @@ pub enum ActionErrorKind {
         #[serde(with = "u128_dec_format")]
         minimum_stake: Balance,
     },
-    /// An error occurred during a `FunctionCall` Action.
-    FunctionCallError(FunctionCallError),
+    /// An error occurred during a `FunctionCall` Action, parameter is debug message.
+    FunctionCallError(String),
```

Comment:
Do not think, that protocol changes in any way, as for protocol it's important presence of FunctionCallError  error, not exact type of it.

Reply:
I think for proof it is important to have an identical hash for the execution outcome, so it will require a protocol change unless I misunderstand something.

Reply:
IIRC, when computing hash we explicitly ignore error messages, @matklad shown me the place where it's done.

Reply:
>@matklad shown me the place where it's done.

Over here: 

https://github.com/near/nearcore/blob/b95eedf81894d87802285cbd331515ee69a0f51f/core/primitives/src/transaction.rs#L324

Reply:
The variant of the error doesn't matter since it doesn't go into the merkle root

Reply:
Great, we don't need a protocol upgrade!

What about database migration?

I also see various ExecutionOutcome types used in chain/network. (ExecutionOutcome uses TxExecutionError which uses ActionError which uses ActionErrorKind and all of that gets borsh-serialized and should be compatible with the existing running nodes, right?)

Reply:
Also do not think we shall store anything related to exact error kind in DB. If we do - guess it's a bug.

Reply:
@olonho we definitely store the errors in the storage since we receive them from RPC: https://explorer.near.org/transactions/GcRiQb5PFwMAogtu8te6wDKMRg9pefbWV4nJx3iqijny

![image](https://user-images.githubusercontent.com/304265/113892834-cd4d0000-979c-11eb-91f3-febcefd1d174.png)

```bash
http post https://archival-rpc.mainnet.near.org/ jsonrpc=2.0 method=tx params:='["GcRiQb5PFwMAogtu8te6wDKMRg9pefbWV4nJx3iqijny", "tom.zest.near"]' id=dontcare
```

Reply:
We do store them in database and if we've changed how the error serializes to borsh we should do a migration

Reply:
Right, so could we do migration in the followup CL? And would be amazing, if we could cooperate on the migration effort.